Assets/Scripts/BlockFaceBehaviour.cs

```diff
@@ -6,116 +6,57 @@
 
 public class BlockFaceBehaviour : MonoBehaviour
 {
+    public float FaceLength;
 
-    public BlockBehaviour Block;
-
+    private BlockBehaviour _block;
     private Color _originalColor;
     private Renderer _renderer;
     private Mesh _mesh;
 
     private Vector3 _normal;
-    private float _faceLength;
 
     private void Start()
     {
-        _renderer = GetComponent<Renderer>();
-        _originalColor = _renderer.material.color;
-
-        MeshFilter meshFilter = GetComponent<MeshFilter>();
-        _mesh = meshFilter.mesh;
-
-        // convert local vertex into world space coordinate
-        Vector3[] worldCoord = GetWorldCoordinates();
-
-        // quad vertices are in this order:
-        // 3 --- 1 
-        // |     | with the normal facing you
-        // 0 --- 2
-        _faceLength = Vector3.Distance(worldCoord[0], worldCoord[2]);
-        _normal = CalculateFaceNormal(worldCoord);
-    }
-
-    public Vector3 GetNormal()
-    {
-        return _normal;
+        _block = GetComponent<BlockBehaviour>();
     }
 
     public float GetFaceLength()
     {
-        return _faceLength;
+        return FaceLength;
     }
 
-    public bool FireRaycastFromFace(float skinToLengthRatio, LayerMask collidableLayers)
+    public bool FireRaycastFromFace(float skinToLengthRatio, LayerMask collidableLayers, BlockFace face)
     {
-        float skinWidth = skinToLengthRatio * _faceLength;   
-
-        // convert local vertex into world space coordinate
-        Vector3[] worldCoord = GetWorldCoordinates();
-
-        // quad vertices are in this order:
-        // 3 -C- 1 
-        // B     D with the normal facing you
-        // 0 -A- 2
-        // 
-        Vector3 a20 = worldCoord[0] - worldCoord[2];
-        Vector3 b03 = worldCoord[3] - worldCoord[0];
-        Vector3 c31 = worldCoord[1] - worldCoord[3];
-        Vector3 d12 = worldCoord[2] - worldCoord[1];
+        float centerToSkin = FaceLength * (1 - skinToLengthRatio / 2) / 2;

```

Comment:
is this supposed to be a ratio or offset? Ratios' value tend to rely on the original value, while offsets should stay constant either way. The way this is written seems to imply it's an offset

Reply:
actually, this is probably right. my bad! >.<